crates/ruff_linter/src/rules/pyupgrade/rules/printf_string_formatting.rs

```diff
@@ -286,6 +291,55 @@ fn clean_params_dictionary(right: &Expr, locator: &Locator, stylist: &Stylist) -
     Some(contents)
 }
 
+/// (In the case where the format string has only a single %-placeholder) and the righthand operand
+/// to % is a variable, try to offer a fix.
+///
+/// If the variable is not a tuple, then offer `({})`.
+/// ```python
+/// x = set(["hi", "hello"])
+/// print("%s" % x)
+/// print("{}".format(x))
+/// ```
+///
+/// If the variable is a 1-tuple, then offer `({}[0])`.
+/// ```python
+/// x = (1,)
+/// print("%s" % x)
+/// print("{}".format(x[0]))
+/// ```
+///
+/// If the variable is an n-tuple, the user has a bug in their code. Offer no fix.
+fn clean_params_variable<'a>(checker: &mut Checker, right: &Expr) -> Option<Cow<'a, str>> {
```

Comment:
I probably misread this, but isn't the method always returning `Cow::Owned`?

Reply:
The method uses `?` syntax to unwrap optionals in at least one place.

Reply:
I think the suggestion is that this can be `Option<String>`, since you aren't using the `Cow::Borrowed` variant.

Reply:
Ok! I was cargo culting the `Cow::Borrowed` constructor. Will fix.

Reply:
Code won't exist in the next version that I push.